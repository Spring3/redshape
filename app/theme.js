export default {
  bg: '#FFFFFF',
  bgLight: '#F9F9F9',
  mainLight: '#FF7F87',
  main: '#FF7079',
  mainDark: '#EF6069',
  minorText: '#909090',
  normalText: '#3F3844',
  hoverText: '#FFFFFF',
  red: '#FF634D',
  yellow: '#FFDA77',
  green: '#6CCA51',
  transitionTime: '.2s'
};
