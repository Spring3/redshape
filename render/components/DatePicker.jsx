import React, { Component } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';

import MenuLeftIcon from 'mdi-react/MenuLeftIcon';
import MenuRightIcon from 'mdi-react/MenuRightIcon';

import { Input } from './Input';

const StyledDatePicker = styled.div`
  .DayPicker {
    display: inline-block;
    font-size: 1rem;
  }

  .DayPicker-wrapper {
    position: relative;
    flex-direction: row;
    padding-bottom: 1em;

    user-select: none;
  }

  .DayPicker-Months {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }

  .DayPicker-Month {
    display: table;
    margin: 0 1em;
    margin-top: 1em;
    border-spacing: 0;
    border-collapse: collapse;

    user-select: none;
  }

  .DayPicker-NavButton {
    position: absolute;
    top: 1em;
    right: 1.5em;
    left: auto;

    display: inline-block;
    margin-top: 2px;
    width: 1.25em;
    height: 1.25em;
    background-position: center;
    background-size: 50%;
    background-repeat: no-repeat;
    cursor: pointer;
  }

  .DayPicker-NavButton:hover {
    opacity: 0.8;
  }

  .DayPicker-NavButton--prev {
    margin-right: 1.5em;
    left: 1.5em;
    right: unset;
    background-image: url('data:image/svg+xml;utf8,${props => props.buttonLeft}');
    background-size: 1.5rem;
  }

  .DayPicker-NavButton--next {
    background-image: url('data:image/svg+xml;utf8,${props => props.buttonRight}');
    background-size: 1.5rem;
  }

  .DayPicker-NavButton--interactionDisabled {
    display: none;
  }

  .DayPicker-Caption {
    display: table-caption;
    margin-bottom: 0.5em;
    padding: 0 0.5em;
    text-align: center;
  }

  .DayPicker-Caption > div {
    font-weight: 500;
    font-size: 1.15em;
  }

  .DayPicker-Weekdays {
    display: table-header-group;
    margin-top: 1em;
  }

  .DayPicker-WeekdaysRow {
    display: table-row;
  }

  .DayPicker-Weekday {
    display: table-cell;
    padding: 0.5em;
    color: ${props => props.theme.normalText};
    font-weight: bold;
    text-align: center;
    font-size: 0.875em;
  }

  .DayPicker-Weekday abbr[title] {
    border-bottom: none;
    text-decoration: none;
  }

  .DayPicker-Body {
    display: table-row-group;
  }

  .DayPicker-Week {
    display: table-row;
  }

  .DayPicker-Day {
    display: table-cell;
    padding: 0.5em;
    border-radius: 50%;
    vertical-align: middle;
    text-align: center;
    cursor: pointer;
  }

  .DayPicker--interactionDisabled .DayPicker-Day {
    cursor: default;
  }

  .DayPicker-Footer {
    padding-top: 0.5em;
  }


  .DayPicker-Day--today {
    color: ${props => props.theme.main};
    font-weight: 700;
  }

  .DayPicker-Day--disabled {
    color: ${props => props.theme.minorText};
    cursor: default;
  }


  .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
    position: relative;
    background-color: ${props => props.theme.normalText};
    color: ${props => props.theme.hoverText};
    font-weight: bold;
  }


  .DayPicker:not(.DayPicker--interactionDisabled)
    .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
      background-color: ${props => props.theme.bgDark};
      color: ${props => props.theme.normalText};
  }

  .DayPickerInput {
    display: inline-block;
  }

  .DayPickerInput-OverlayWrapper {
    position: relative;
    left: -30%;
    bottom: -5px;
  }

  .DayPickerInput-Overlay {
    position: absolute;
    left: 0;
    z-index: 1;

    background: ${props => props.theme.bg};
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
  }
`;

class DatePicker extends Component {
  render() {
    const { theme, value, name, isDisabled, onChange } = this.props;
    return (
      <StyledDatePicker
        buttonLeft={encodeURIComponent(renderToStaticMarkup(<MenuLeftIcon xmlns="http://www.w3.org/2000/svg" color={theme.normalText} />))}
        buttonRight={encodeURIComponent(renderToStaticMarkup(<MenuRightIcon xmlns="http://www.w3.org/2000/svg" color={theme.normalText} />))}
      >
        <DayPickerInput
          styles={this.styles}
          name={name}
          value={value}
          inputProps={{
            disabled: isDisabled
          }}
          disabled={isDisabled}
          placeholder={false}
          onDayChange={onChange}
          component={props => (<Input {...props} />)}
        />
      </StyledDatePicker>
    );
  }
}

DatePicker.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.instanceOf(Date),
    PropTypes.string
  ]),
  isDisabled: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  theme: PropTypes.object.isRequired
};

DatePicker.defaultProps = {
  value: new Date(),
  isDisabled: false,
  name: null
};

export default withTheme(DatePicker);
