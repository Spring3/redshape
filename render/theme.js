export default {
  bg: '#FFFFFF',
  bgDark: '#F9F9F9',
  bgDarker: '#E9E9E9',
  bgDisabled: 'hsl(0,0%,95%)',
  mainLight: '#FF7F87',
  main: '#FF7079',
  mainDark: '#EF6069',
  minorText: '#A0A0A0',
  normalText: '#3F3844',
  hoverText: '#FFFFFF',
  red: '#FF634D',
  yellow: '#FFDA77',
  green: '#6CCA51',
  shadow: '#D0D0D0',
  transitionTime: '.2s'
};
