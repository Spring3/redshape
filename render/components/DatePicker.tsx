import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import { useTheme } from 'styled-components';
import { css } from '@emotion/react';
import { DayModifiers } from 'react-day-picker';

import MenuLeftIcon from 'mdi-react/MenuLeftIcon';
import MenuRightIcon from 'mdi-react/MenuRightIcon';

import { Input } from './Input';
import { theme as Theme } from '../theme';

const styles = ({
  buttonLeft,
  buttonRight,
  theme
}: {
  buttonLeft: string;
  buttonRight: string;
  theme: typeof Theme;
}) => css`
  .DayPicker {
    display: inline-block;
    font-size: 1rem;
  }

  .DayPicker-wrapper {
    position: relative;
    flex-direction: row;
    padding-bottom: 1em;

    user-select: none;
  }

  .DayPicker-Months {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }

  .DayPicker-Month {
    display: table;
    margin: 0 1em;
    margin-top: 1em;
    border-spacing: 0;
    border-collapse: collapse;

    user-select: none;
  }

  .DayPicker-NavButton {
    position: absolute;
    top: 1em;
    right: 1.5em;
    left: auto;

    display: inline-block;
    margin-top: 2px;
    width: 1.25em;
    height: 1.25em;
    background-position: center;
    background-size: 50%;
    background-repeat: no-repeat;
    cursor: pointer;
  }

  .DayPicker-NavButton:hover {
    opacity: 0.8;
  }

  .DayPicker-NavButton--prev {
    margin-right: 1.5em;
    left: 1.5em;
    right: unset;
    background-image: url('data:image/svg+xml;utf8,${buttonLeft}');
    background-size: 1.5rem;
  }

  .DayPicker-NavButton--next {
    background-image: url('data:image/svg+xml;utf8,${buttonRight}');
    background-size: 1.5rem;
  }

  .DayPicker-NavButton--interactionDisabled {
    display: none;
  }

  .DayPicker-Caption {
    display: table-caption;
    margin-bottom: 0.5em;
    padding: 0 0.5em;
    text-align: center;
  }

  .DayPicker-Caption > div {
    font-weight: 500;
    font-size: 1.15em;
  }

  .DayPicker-Weekdays {
    display: table-header-group;
    margin-top: 1em;
  }

  .DayPicker-WeekdaysRow {
    display: table-row;
  }

  .DayPicker-Weekday {
    display: table-cell;
    padding: 0.5em;
    color: ${theme.normalText};
    font-weight: bold;
    text-align: center;
    font-size: 0.875em;
  }

  .DayPicker-Weekday abbr[title] {
    border-bottom: none;
    text-decoration: none;
  }

  .DayPicker-Body {
    display: table-row-group;
  }

  .DayPicker-Week {
    display: table-row;
  }

  .DayPicker-Day {
    display: table-cell;
    padding: 0.5em;
    border-radius: 50%;
    vertical-align: middle;
    text-align: center;
    cursor: pointer;
  }

  .DayPicker--interactionDisabled .DayPicker-Day {
    cursor: default;
  }

  .DayPicker-Footer {
    padding-top: 0.5em;
  }


  .DayPicker-Day--today {
    color: ${theme.main};
    font-weight: 700;
  }

  .DayPicker-Day--disabled {
    color: ${theme.minorText};
    cursor: default;
  }


  .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
    position: relative;
    background-color: ${theme.normalText};
    color: ${theme.hoverText};
    font-weight: bold;
  }


  .DayPicker:not(.DayPicker--interactionDisabled)
    .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
      background-color: ${theme.bgDark};
      color: ${theme.normalText};
  }

  .DayPickerInput {
    display: inline-block;
  }

  .DayPickerInput-OverlayWrapper {
    position: relative;
    left: -30%;
    bottom: -5px;
  }

  .DayPickerInput-Overlay {
    position: absolute;
    left: 0;
    z-index: 1;

    background: ${theme.bg};
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
  }
`;

type DatePickerProps = {
  id?: string;
  value?: string;
  name?: string;
  disabled?: boolean;
  onChange?: (day: Date, dayModifiers: DayModifiers, dayPickerInput: DayPickerInput) => void;
}

const DatePicker = ({ id, value, disabled = false, onChange }: DatePickerProps) => {
  const theme = useTheme() as typeof Theme;
  const buttonLeft = encodeURIComponent(
    renderToStaticMarkup(
      <MenuLeftIcon xmlns="http://www.w3.org/2000/svg" color={theme.normalText} />
    )
  );
  const buttonRight = encodeURIComponent(
    renderToStaticMarkup(
      <MenuRightIcon xmlns="http://www.w3.org/2000/svg" color={theme.normalText} />
    )
  );
  return (
    <div id={id} css={styles({ buttonLeft, buttonRight, theme })}>
      <DayPickerInput
        value={value}
        placeholder='yyyy-MM-dd'
        inputProps={{
          disabled
        }}
        onDayChange={onChange}
        component={(props: any) => <Input maxWidth='150px' {...props} />}
      />
    </div>
  );
};

export {
  DatePicker
};
