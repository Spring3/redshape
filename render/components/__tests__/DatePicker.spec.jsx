import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import DatePicker from '../DatePicker';
import theme from '../../theme';

describe('Date Picker', () => {
  afterEach(cleanup);
  it('should render the date picker', () => {
    render(
      <DatePicker
        theme={theme}
        value={123456789}
        onChange={() => {}}
      />
    );
    const input = document.querySelector('input');
    expect(input).toHaveAttribute('value', '123456789');
  });
});
