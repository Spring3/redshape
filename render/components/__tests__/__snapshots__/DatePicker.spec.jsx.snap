// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Picker should match the snapshot 1`] = `
.c1 {
  display: block;
  width: 100%;
  border-radius: 3px;
  padding: 5px 10px;
  box-sizing: border-box;
  font-size: 14px;
  min-height: 35px;
  outline: none;
  font-weight: bold;
  -webkit-transition: background;
  transition: background;
  border: 1px solid;
  background: white;
}

.c1:hover {
  border: 1px solid;
}

.c1:focus {
  box-shadow: 0px 0px 0px 1px;
}

.c1::-webkit-input-placeholder {
  font-weight: 500;
}

.c1::-moz-placeholder {
  font-weight: 500;
}

.c1:-ms-input-placeholder {
  font-weight: 500;
}

.c1::placeholder {
  font-weight: 500;
}

.c0 .DayPicker {
  display: inline-block;
  font-size: 1rem;
}

.c0 .DayPicker-wrapper {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-bottom: 1em;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .DayPicker-Months {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .DayPicker-Month {
  display: table;
  margin: 0 1em;
  margin-top: 1em;
  border-spacing: 0;
  border-collapse: collapse;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .DayPicker-NavButton {
  position: absolute;
  top: 1em;
  right: 1.5em;
  left: auto;
  display: inline-block;
  margin-top: 2px;
  width: 1.25em;
  height: 1.25em;
  background-position: center;
  background-size: 50%;
  background-repeat: no-repeat;
  cursor: pointer;
}

.c0 .DayPicker-NavButton:hover {
  opacity: 0.8;
}

.c0 .DayPicker-NavButton--prev {
  margin-right: 1.5em;
  left: 1.5em;
  right: unset;
  background-image: url('data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20class%3D%22mdi-icon%20%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%233F3844%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22M14%2C7L9%2C12L14%2C17V7Z%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E');
  background-size: 1.5rem;
}

.c0 .DayPicker-NavButton--next {
  background-image: url('data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20class%3D%22mdi-icon%20%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%233F3844%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22M10%2C17L15%2C12L10%2C7V17Z%22%3E%3C%2Fpath%3E%3C%2Fsvg%3E');
  background-size: 1.5rem;
}

.c0 .DayPicker-NavButton--interactionDisabled {
  display: none;
}

.c0 .DayPicker-Caption {
  display: table-caption;
  margin-bottom: 0.5em;
  padding: 0 0.5em;
  text-align: center;
}

.c0 .DayPicker-Caption > div {
  font-weight: 500;
  font-size: 1.15em;
}

.c0 .DayPicker-Weekdays {
  display: table-header-group;
  margin-top: 1em;
}

.c0 .DayPicker-WeekdaysRow {
  display: table-row;
}

.c0 .DayPicker-Weekday {
  display: table-cell;
  padding: 0.5em;
  font-weight: bold;
  text-align: center;
  font-size: 0.875em;
}

.c0 .DayPicker-Weekday abbr[title] {
  border-bottom: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .DayPicker-Body {
  display: table-row-group;
}

.c0 .DayPicker-Week {
  display: table-row;
}

.c0 .DayPicker-Day {
  display: table-cell;
  padding: 0.5em;
  border-radius: 50%;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
}

.c0 .DayPicker--interactionDisabled .DayPicker-Day {
  cursor: default;
}

.c0 .DayPicker-Footer {
  padding-top: 0.5em;
}

.c0 .DayPicker-Day--today {
  font-weight: 700;
}

.c0 .DayPicker-Day--disabled {
  cursor: default;
}

.c0 .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
  position: relative;
  font-weight: bold;
}

.c0 .DayPickerInput {
  display: inline-block;
}

.c0 .DayPickerInput-OverlayWrapper {
  position: relative;
  left: -30%;
  bottom: -5px;
}

.c0 .DayPickerInput-Overlay {
  position: absolute;
  left: 0;
  z-index: 1;
  box-shadow: 0 2px 5px rgba(0,0,0,0.15);
}

<div
  className="c0"
>
  <div
    className="DayPickerInput"
  >
    <input
      className="c1"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      placeholder={false}
      type="text"
      value={123456789}
    />
  </div>
</div>
`;
